engine/src/core/kstring.c

```diff
@@ -234,6 +234,29 @@ i32 string_index_of(const char* str, char c) {
     return -1;
 }
 
+i32 string_index_of_str(const char* str_0, const char* str_1) {
+    if (!str_0 || !str_1) {
+        return -1;
+    }
+    u32 length_0 = string_length(str_0);
+    u32 length_1 = string_length(str_1);
+    if (length_0 > 0 && length_1 > 0) {
```

Comment:
Technically speaking an *empty string* is present in any other string. The value of `0` should be returned in such a case. The rationale goes as follows:

If a string A is a substring of string B then dropping any *trailing* character from S1 should preserve inclusion.